### PR TITLE
8253240: No javadoc for DecimalFormatSymbols.hashCode()

### PR DESCRIPTION
**Pull request by naotoj**
Hi,

Please review the fix to the issue wrt missing hashCode() javadoc, which was recently discussed in core-libs ml.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8253240](https://bugs.openjdk.java.net/browse/JDK-8253240): No javadoc for DecimalFormatSymbols.hashCode()

### Reviewers
 * [Roger Riggs](https://openjdk.java.net/census#rriggs) (@RogerRiggs - **Reviewer**)
 * [Lance Andersen](https://openjdk.java.net/census#lancea) (@LanceAndersen - **Reviewer**) ⚠️ Review applies to 3276598e992ebc598150b150041859bebb8a3daf

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/208/head:pull/208`
`$ git checkout pull/208`

### CODE REVIEW

File: src/java.base/share/classes/java/text/DecimalFormatSymbols.java [hunk 1]
```diff
@@ -1148,6 +1147,11 @@ private void readObject(ObjectInputStream stream)
     private transient Currency currency;
     private transient volatile boolean currencyInitialized;
 
+    /**
+     * Cached hash code
+     */
+    private volatile int hashCode;
```
Comment (RogerRiggs) [inline src/java.base/share/classes/java/text/DecimalFormatSymbols.java:1153]: I'm thinking the cached hashcode should be / should have been transient (not serialized).
And add a period at the end of the sentence above.

**Review by RogerRiggs — CHANGES_REQUESTED** [review]

**Review by LanceAndersen — APPROVED** [review]
Looks in line with the discussion on core-libs.

File: src/java.base/share/classes/java/text/DecimalFormatSymbols.java (hunk 1 above)
Comment (naotoj) [reply src/java.base/share/classes/java/text/DecimalFormatSymbols.java]: Good point. Addressed both comments.

**Comment by RogerRiggs** [comment]
Removing hashcode from the serialized form needs a CSR because it was non-transient in JDK 15.
The intent is to not use the serialized value, so the readObject method should zero it out
so it can be computed on first use.

/csr needed

**Comment by naotoj** [comment]
Drafted a CSR for this PR. Please review.

**Review by RogerRiggs — APPROVED** [review]
Looks fine.  Thanks

**Pull request closed without merging**